synapse/storage/databases/main/state.py

```diff
@@ -485,6 +486,7 @@ def _update_state_for_partial_state_event_txn(
                 "rejection_status_changed": rejection_status_changed,
             },
         )
+        txn.call_after(self.hs.get_notifier().on_new_replication_data)
```

Comment:
I'm confused about when to use `on_new_replication_data()` and `notify_replication()`.

In a previous PR, we added a call to `notify_replication` after calling `update_state_for_partial_state_event`. Is that not sufficient to have replication work?
I do prefer poking replication much closer to the db transaction though, except we have a precedent of calling `notify_replication` outside db transactions.

https://github.com/matrix-org/synapse/blob/develop/synapse/handlers/federation_event.py#L609-L614

Reply:
`on_new_replication_data` appear to be an alias of `notify_replication` :/



> In a previous PR, we added a call to `notify_replication` after calling `update_state_for_partial_state_event`. Is that not sufficient to have replication work?

Note that that might poke replication on a *different* worker, as we dispatch `update_state` to the master

Reply:
> Note that that might poke replication on a different worker, as we dispatch `update_state` to the master

Darn, I missed that completely when reviewing #14545.